code-challenges.test.js

```diff
@@ -12,40 +21,150 @@
+  it("takes in a mixed data array and returns an array of only the REMAINDERS of the numbers when divided by 3.", () => {
+    const hodgepodge1 = [23, "Heyyyy!", 45, -10, 0, "Yo", false]
+    // Expected output: [ 2, 0, -1, 0 ]
+    const hodgepodge2 = [5, "Hola", 43, -34, "greetings", true]
+    // Expected output: [ 2, 1, -1 ]
+    expect(onlyRemainD3(hodgepodge1)).toEqual([ 2, 0, -1, 0 ])
+    expect(onlyRemainD3(hodgepodge2)).toEqual([ 2, 1, -1 ])
+  })
+})
+// ● onlyRemainD3 › takes in a mixed data array and returns an array of only the REMAINDERS of the numbers when divided by 3.
+
+// ReferenceError: onlyRemainD3 is not defined
 
 
 // b) Create the function that makes the test pass.
+// create a  function called onlyRemainD3
+//use filter to remove non numbers
+//within that filter I think I can make it return the remainders of the numbers so lets see
+
+const onlyRemainD3 = (threeLittlePigs) => {
+  return threeLittlePigs.filter(value => {
+    return typeof value === "number" 
+  }).map(value => {
+    return value % 3
+  })
+}
+
+//had to split work it into a map could not get the math to work within the filter.
+// √ takes in a mixed data array and returns an array of only the REMAINDERS of the numbers when divided by 3.
+
+// Test Suites: 1 passed, 1 total
+// Tests:       2 passed, 2 total
 
 
 // --------------------3) Create a function that takes in an array of numbers and returns the sum of all the numbers cubed.
 
+//Follow Red, Green, Refactor
+//describe,it,expect,toEqual
+//describe cubeNSum
+//it  takes in an array of numbers and returns the sum of all the numbers cubed.
+//expect (cubeNSum(cubeAndSum1))
+//toEqual 99
+
 // a) Create a test with an expect statement using the variables provided.
 
-const cubeAndSum1 = [2, 3, 4]
-// Expected output: 99
-const cubeAndSum2 = [0, 5, 10]
-// Expected output: 1125
+describe("cubeNSum",() => {
+  it("takes in an array of numbers and returns the sum of all the numbers cubed.",() => {
+    const cubeAndSum1 = [2, 3, 4]
+    // Expected output: 99
+    const cubeAndSum2 = [0, 5, 10]
+    // Expected output: 1125
+    expect(cubeNSum(cubeAndSum1)).toEqual(99)
+    expect(cubeNSum(cubeAndSum2)).toEqual(1125)
+  })
+})
+// cubeNSum
+// × takes in an array of numbers and returns the sum of all the numbers cubed.
+
+// ● cubeNSum › takes in an array of numbers and returns the sum of all the numbers cubed.
+
+// ReferenceError: cubeNSum is not defined
 
 
 // b) Create the function that makes the test pass.
+
+//create function called cubeNSum
+//use a loop to add numbers
+//set up a variable to hold prior sum
+//return prior sum at the end of the loop
+//during the loop cube the numbers prior to summing
+
+const cubeNSum = (musketeers) => {
+  let priorSum = 0
+  for(let i = 0; i < musketeers.length; i++){
+    let currentCube = musketeers[i]*musketeers[i]*musketeers[i]
+    let currentSum = currentCube + priorSum
+    priorSum = currentSum
+  }
+  return priorSum
+}
```

Comment:
Logic looks great!  Can you think of how to refactor this using higher order functions?